config/core/configmaps/autoscaler.yaml

```
@@ -187,3 +187,9 @@ data:
     # unless overridden by the "autoscaling.knative.dev/maxScale" annotation.
     # If set to 0, the revision has no maximum scale.
     max-scale: "0"
+
+    # max-scale-limit ensures that both cluster-wide flag max-scale and per-revision
+    # annotation "autoscaling.knative.dev/maxScale" for new revision will not exceed
+    # this number. If max-scale-limit is greater than 0, it disables both explicitly
+    # setting "autoscaling.knative.dev/maxScale" annotation to 0 and leaving it unset.
```

I feel like the fact that this affects both the annotation and the config might be implicit. How about:

```suggestion
    # max-scale-limit sets the maximum permitted value for the max scale of a revision.
    # When this is set to a positive value, a revision with a maxScale above that value
    # (including a maxScale of "0" = unlimited) is disallowed.
    # A value of zero (the default) allows any limit, including unlimited.
```